object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -573,8 +573,7 @@ private void handleSuccessorBlocks(
             || setAssign.stream()
                 .allMatch(assign -> succRegularStore.getValue(assign.localVar) == null)) {
           if (nodes.size() == 0) { // If the cur block is special or conditional block
-            checkMustCall(setAssign, succRegularStore, reasonForSucc);
-
+            checkMustCall(setAssign, analysis.getInput(block).getRegularStore(), reasonForSucc);
```

Comment:
This change makes sense to me in the case where none of the local variables in `setAssign` appear in `succRegularStore`.  In such a case, the variable is going out of scope, and it doesn't make sense to pass `succRegularStore` to `checkMustCall`.

Now let's consider the case where some local variable _does_ appear in `succRegularStore`.  There, we have the question of whether some variable type in `succRegularStore` could differ from its type in `analysis.getInput(block).getRegularStore()`.  But since we know `nodes.size() == 0`, I _think_ this is not possible.

At the least, can we add a comment explaining why we are using the block we choose here, maybe with some of the argument above?  We could also add some logic to only pass `analysis.getInput(block).getRegularStore()` when the `setAssign.stream().allMatch(...)` expression above is `true`.  But I don't think that should be necessary.

Reply:
> We could also add some logic to only pass analysis.getInput(block).getRegularStore() when the setAssign.stream().allMatch(...) expression above is true.

Doing this fixes the problem related to `EnsuresCalledMethodsIf` I described below. I've pushed that version of the code.